.github/workflows/pull_upstream_with_cherrypick.yml

```diff
@@ -0,0 +1,100 @@
+name: Pull from Upstream and Cherry-pick Patches
+
+on:
+  workflow_dispatch:
+    inputs:
+      upstream_repo:
+        description: 'Select Upstream Repository'
+        required: true
+        type: choice
+        options:
+          - 'openvino_model_server'
+          - 'kserve'
+          - 'modelmesh'
+          - 'caikit-tgis-serving'
+          - 'openvino'
+          - 'vllm'
+          - 'caikit-nlp'
+          - 'caikit'
+          - 'odh-model-controller'
+          - 'caikit-tgis-backend'
+          - 'caikit-nlp-client'
+          - 'model-registry'
+      upstream_branch:
+        description: 'Upstream branch to pull from'
+        required: true
+      target_branch:
+        description: 'Target branch to pull into'
+        required: true
+      patch_commits:
+        description: 'Comma-separated list of commit SHAs to cherry-pick'
+        required: true
+
+permissions:
+  contents: write
+  packages: write
+  pull-requests: write
+
+jobs:
+  pull_and_cherry_pick:
+    runs-on: ubuntu-latest
+    outputs: 
+        upstream_repo: ${{ steps.set-repo.outputs.upstream_org_repo }}
+        midstream_repo: ${{ steps.set-repo.outputs.midstream_org_repo }}
+    env:
+      BASE_UPSTREAM_URL: https://github.com/opendatahub-io/
+      BASE_TARGET_URL: https://github.com/red-hat-data-services/
+      
+    steps:
+    - name: Set repository
+      id: set-repo
+      run: |
+        echo "upstream_org_repo=opendatahub-io/${{ github.event.inputs.upstream_repo }}" >> $GITHUB_OUTPUT
+        echo "midstream_org_repo=red-hat-data-services/${{ github.event.inputs.upstream_repo }}" >> $GITHUB_OUTPUT

```

Comment:
opendatahub-io is midstream, red-hat-data-services is a downstream

In this case, the intent of the workflow is to take from upstream, and cherry-pick a series of commits on top of it, then merge that to a target repo+branch